### PR TITLE
Specify YAML coder for mdm payload

### PR DESCRIPTION
**Pull request by dwelch-r7**
We need to specify a coder here to be compatible with Rails 7.1

Tested as part of the framework PR rails 7.1 here: https://github.com/rapid7/metasploit-framework/pull/19953
Rails 7.0 here: https://github.com/rapid7/metasploit-framework/pull/19977

### CODE REVIEW

**Review by cgranleese-r7 — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ group :development, :test do
   # auto-load factories from spec/factories
   gem 'factory_bot_rails'
 
-  gem 'rails', '~>  7.0'
+  gem 'rails', '~>  7.1.0'
```
Comment (adfoster-r7) [inline Gemfile:22]: We still want to test 7.0.x here - since that's what will actually be shipped to customers

```suggestion
  gem 'rails', '~> 7.0.8'
```

We might want to lock in a better pattern moving forward though, since this kind of edge-case will happen again in the future. i.e. we should really be testing against multiple versions of Rails, so thata we can detect issues sooner?

File: db/migrate/20110317144932_add_session_table.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class AddSessionTable < ActiveRecord::Migration[4.2]
 
 	class Event < ApplicationRecord
-		serialize :info
+		serialize :info, coder: YAML
```
Comment (adfoster-r7) [inline db/migrate/20110317144932_add_session_table.rb:4]: Can we temporarily follow the existing pattern - even if it's not super required here, more so for future travellers that might be doing rails upgrades here

File: db/migrate/20110317144932_add_session_table.rb [hunk 3]
```diff
@@ -1,7 +1,11 @@
 class AddSessionTable < ActiveRecord::Migration[4.2]
 
 	class Event < ApplicationRecord
-		serialize :info
+		if ActiveRecord::VERSION::MAJOR >= 7 && ActiveRecord::VERSION::MINOR >= 1
```
Comment (dwelch-r7) [inline db/migrate/20110317144932_add_session_table.rb:4]: we're tab indented here for some reason, ignore the formatting it looks normal in the IDE, file was orinigally tabs so not going to change that

File: app/models/mdm/payload.rb [hunk 4]
```diff
@@ -92,10 +92,18 @@ class Mdm::Payload < ApplicationRecord
...
   # Serializations
-  #
 
-  serialize :urls
-  serialize :build_opts
+  if ActiveRecord::VERSION::MAJOR >= 7 && ActiveRecord::VERSION::MINOR >= 1
+    serialize :urls, coder: YAML
+  else
+    serialize :urls
+  end
```
Comment (dwelch-r7) [inline app/models/mdm/payload.rb:100]: since specifying `coder:` is a noop, and the old code didn't specify _any_ coder, this `if/else` statementi s kind of redundant, I did add in an explicit `serialize :urls, YAML` but that broke the tests and it makes more sense to keep the old functionality the same as it was before
I don't mind keeping it here, since it makes it obvious what the intention is

**Review by adfoster-r7 — APPROVED** [review]
